### ISSUE TITLE
Stochastic Gradient Descent

### ISSUE THREAD

**Issue by danmcleran** [danmcleran/tinymind#36]
Look into efficient stochastic methods in lieu of backpropagation via gradient descent. e.g. Adam https://arxiv.org/pdf/1412.6980.pdf
